packages/flutter/lib/src/material/icon_button.dart

```diff
@@ -893,13 +924,13 @@ class _SelectableIconButtonState extends State<_SelectableIconButton> {
   void didUpdateWidget(_SelectableIconButton oldWidget) {
     super.didUpdateWidget(oldWidget);
     if (widget.isSelected == null) {
-      if (statesController.value.contains(MaterialState.selected)) {
-        statesController.update(MaterialState.selected, false);
+      if (_internalStatesController.value.contains(MaterialState.selected)) {
```

Comment:
Should we be using `statesController` instead of `_internalStatesController` directly here and usages below in this file so the changes are reflected in a user provided controller?